### PR TITLE
Add cookbook example for growing and shrinking a pool

### PR DESCRIPTION
**Pull request by davidbeckingsale**

### CODE REVIEW

File: examples/cookbook/recipe_one_shrink.cpp [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+#include <iostream>
+
+int main(int, char**) {
+  auto& rm = umpire::ResourceManager::getInstance();
+
+  auto allocator = rm.getAllocator("DEVICE");
+
+  /* 
+   * Create a 4 Gb pool and reserve one byte.
```
Comment (mcfadden8) [inline examples/cookbook/recipe_one_shrink.cpp]: You probably already know this, but the actual implementation will allocate more than just one byte for alignment purposes.

Comment (davidbeckingsale) [reply examples/cookbook/recipe_one_shrink.cpp]: Yes, I ought to change it to be explicitly a word?

**Comment by davidbeckingsale** [comment]
@mcfadden8 - this should be ready to roll.

**Comment by davidbeckingsale** [comment]
@mcfadden8 - can I get a re-approval, the build was failing with an "unused variable" warning.

**Pull request merged**